vllm/attention/ops/triton_flash_attention.py

```diff
@@ -207,103 +209,186 @@ def _attn_fwd_inner(
     return acc, l_i, m_i
 
 
-@triton.autotune(
-    configs=[
+def get_gfx_version():
+    try:
+        # Run the rocminfo command
+        result = subprocess.run(['rocminfo'],
+                                stdout=subprocess.PIPE,
+                                stderr=subprocess.PIPE,
+                                text=True)
+        output = result.stdout
+
+        # Parse the output to find the gfx version
+        for line in output.splitlines():
+            line = line.strip()
+            if line.startswith("Name: gfx"):
+                gfx_version = line.split("Name:")[1].strip()
+                return gfx_version
+    except Exception as e:
+        print(f"Error: {e}")
+    return None
+
+
+def is_hip():
+    return triton.runtime.driver.active.get_current_target().backend == "hip"
+
+
+def is_cdna():
+    return is_hip() and triton.runtime.driver.active.get_current_target(
+    ).arch in ('gfx940', 'gfx941', 'gfx942', 'gfx90a', 'gfx908')
+
+
+def is_rdna():
```

Comment:
probably worth to use: https://github.com/ROCm/vllm/blob/8f3bf8b2b5bf1d231007629b3a5fbbf235f0cfe9/vllm/utils.py#L1620

Reply:
updated